### PR TITLE
Order

### PR DESCRIPTION
**Pull request by duccm94**

### CODE REVIEW

File: app/controllers/order_details_controller.rb [hunk 1]
```diff
@@ -0,0 +1,76 @@
+class OrderDetailsController < ApplicationController
+  before_action :check_current_order, except: [:show, :new]
+  before_action :find_order_detail, only: [:edit, :update, :destroy]
+  before_action :check_product_exist_in_cart, only: [:create]
```
Comment (ngtrduc) [inline app/controllers/order_details_controller.rb]: bo []

File: app/controllers/order_details_controller.rb [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+      flash[:success] = t "success"
+      redirect_to order_path
+    else
+      flash[:danger] = t "error"
+      redirect_to :edit
+    end
+  end
+
+  def destroy
+    @order_detail.destroy
```
Comment (ngtrduc) [inline app/controllers/order_details_controller.rb]: thua

File: app/controllers/order_details_controller.rb [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    end
+  end
+
+  private
+  def order_detail_params
+    params.require(:order_detail).permit :quantity, :product_id
+  end
+
+  def find_order_detail
+    @order_detail = @order.order_details.find params[:id]
```
Comment (ngtrduc) [inline app/controllers/order_details_controller.rb]: dùng find_by_id

**Comment by VuThiHien94** [comment]
OK

**Comment by daonhat** [comment]
ok

File: app/views/orders/show.html.erb [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    <% end %>
+    <button type="button" class="btn btn-primary" data-toggle="modal"
+      data-target="#order_coin"><%= t "order.coin" %></button>
+    <%= form_for @order do |f| %>
+      <div class="modal fade" id="order_coin" tabindex="-1" role="dialog"
+        aria-labelledby="myModalLabel">
+        <div class="modal-dialog" role="document">
+          <div class="modal-content">
+            <div class="modal-header">
+              <button type="button" class="close" data-dismiss="modal" aria-label="Close">
```
Comment (ngtrduc) [inline app/views/orders/show.html.erb]: i18n

**Comment by ngtrduc** [comment]
ok

File: app/controllers/application_controller.rb [hunk 5]
```diff
@@ -3,4 +3,14 @@ class ApplicationController < ActionController::Base
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
   before_action :authenticate_user!
+  helper_method :current_order
+
+  def current_order
+    if user_signed_in?
```
Comment (hellovietnam93) [inline app/controllers/application_controller.rb]: @current_order = if .. else .. end

**Pull request merged**